src/reader.cpp

```diff
@@ -168,6 +168,8 @@ class reader_data_t {
     bool suppress_autosuggestion;
     /// Whether abbreviations are expanded.
     bool expand_abbreviations;
+    /// Silent mode used for password input on the read command
+    bool silent;
```

Comment:
I don't see where you are initializing this in the constructor for the class. Which means the initial value it undefined. See the `/// Constructor` comment a hundred lines below.